lib/assert.js

```diff
@@ -95,43 +93,21 @@ function innerFail(obj) {
 }
 
 /**
- * @param {any} actual
- * @param {any} expected
- * @param {string | Error} [message]
- * @param {string} [operator]
- * @param {Function} [stackStartFn]
+ * Throws an AssertionError with the given message.
+ * @param {any | Error} [message]
  */
-function fail(actual, expected, message, operator, stackStartFn) {
-  const argsLen = arguments.length;
+function fail(message) {
+  if (isError(message)) throw message;
```

Comment:
Nit: we could move this down as else case to the `message == null` check. It's not really important here, since that won't make a real difference, it just feels right for me :)